app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -943,38 +950,93 @@ class BrowserTabFragment :
         }
     }
 
+    /**
+     * This method prevents the toolbar from overlapping the content of the page when the page is not scrollable.
+     *
+     * It checks if the page is scrollable and if it is, it makes the bottom toolbar
+     * collapsible. If the page is not scrollable, it makes the bottom toolbar always visible and adjusts the bottom padding of the WebView by the
+     * toolbar height.
+     *
+     */
+    private fun makeBottomOmnibarStickyIfNeeded() {
+        webView?.let { duckDuckGoWebView ->
+            lifecycleScope.launch {
+                checkIfSiteScrollable(duckDuckGoWebView)
+                delay(SCROLLABILITY_CHECK_DELAY) // delay added to ensure the site had a chance to settle before checking one more time
+                checkIfSiteScrollable(duckDuckGoWebView)
+            }
+        }
+    }
+
+    private suspend fun checkIfSiteScrollable(duckDuckGoWebView: DuckDuckGoWebView) {
```

Comment:
nit: this method not only checks for scrollable, it also changes paddings and layout params. The method name should reflect that too.